sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -646,11 +646,64 @@ internal virtual Task ClearAsync(CancellationToken cancellationToken = default)
         ///   This method will automatically unregister the <see cref="SendEventBatchSucceededAsync"/> and <see cref="SendEventBatchFailedAsync"/> handlers.
         /// </remarks>
         ///
-        public virtual Task CloseAsync(bool flush = true,
-                                       CancellationToken cancellationToken = default)
+        public virtual async Task CloseAsync(bool flush = true,
+                                             CancellationToken cancellationToken = default)
         {
+            if (_isClosed)
```

Comment:
Is the `else` path ok to run twice in the event of a thread race?

Reply:
damn.  Good catch, it is not.